File: internal/api/dto/favorite.go
package dto

// AddFavoriteReq 添加收藏请求。
type AddFavoriteReq struct {
	Title string  `json:"title" binding:"required"`
	URL   string  `json:"url" binding:"required"`
	Icon  *string `json:"icon"`
}

// ImportFavoritesReq 导入收藏请求。
type ImportFavoritesReq struct {
	Favorites []AddFavoriteReq `json:"favorites" binding:"required"`
}

File: apps/ui/src/api/favorite.ts
import { http } from "@/utils";

export const getFavoritesKey = "/api/favorites";
export const getFavorites = () => http.get(getFavoritesKey);

export const addFavorite = (fav: {
  title: string;
  url: string;
  icon?: string;
}) => http.post("/api/favorites", fav);

export const removeFavorite = (id: number) =>
  http.delete(`/api/favorites/${id}`);

export const exportFavorites = () => http.get("/api/favorites/export");

export const importFavorites = (favorites: unknown[]) =>
  http.post("/api/favorites/import", { favorites });
